test/generate_ssl_tests.pl

```diff
@@ -127,17 +127,28 @@ sub print_templates {
 # Shamelessly copied from Configure.
 sub read_config {
     my $fname = shift;
+    my $provider = shift;
+    my $fips_mode = "0";
+    my $no_deflt_libctx = "0";
+
+    $fips_mode = "1" if $provider eq "fips";
+    $no_deflt_libctx = "1" if $provider eq "default" || $provider eq "fips";
+
     open(INPUT, "< $fname") or die "Can't open input file '$fname'!\n";
     local $/ = undef;
     my $content = <INPUT>;
+    $content =~ s/FIPS_MODE/$fips_mode/g;
+    $content =~ s/NO_DEFLT_LIBCTX/$no_deflt_libctx/g;
```

Comment:
Ewwww.....

May I suggest this instead, then just use `$fips_mode` and `$no_deflt_libctx` directly in the `.cnf.in` files?

``` diff
diff --git a/test/generate_ssl_tests.pl b/test/generate_ssl_tests.pl
index 8cfc451fbb..ce8c22f59c 100644
--- a/test/generate_ssl_tests.pl
+++ b/test/generate_ssl_tests.pl
@@ -127,6 +127,9 @@ sub print_templates {
 # Shamelessly copied from Configure.
 sub read_config {
     my $fname = shift;
+    local $ssltests::fips_mode = $provider eq "fips";
+    local $ssltests::no_deflt_libctx =
+        $provider eq "default" || $provider eq "fips";
     open(INPUT, "< $fname") or die "Can't open input file '$fname'!\n";
     local $/ = undef;
     my $content = <INPUT>;
diff --git a/test/ssl-tests/ssltests_base.pm b/test/ssl-tests/ssltests_base.pm
index c6004fa7b2..98c9a839ad 100644
--- a/test/ssl-tests/ssltests_base.pm
+++ b/test/ssl-tests/ssltests_base.pm
@@ -17,6 +17,9 @@ sub test_pem
     return "\${ENV::TEST_CERTS_DIR}" . $dir_sep . $file,
 }
 
+our $fips_mode = 0;
+our $no_deflt_libctx = 0;
+
 our %base_server = (
     "Certificate" => test_pem("servercert.pem"),
     "PrivateKey"  => test_pem("serverkey.pem"),
```

Reply:
Thanks - I've included that patch.